### DISCUSSION TITLE
MQTT client vs SDK IOT client

### DISCUSSION THREAD

**Discussion by markpope** [Azure/azure-iot-sdk-java#1631] [General]
Clearly new to IOT here, what are the benefits of using a SDK IOT client over an MQTT client publishing to port 8883 on an IOT Hub?

**Comment by timtay-microsoft** [comment]
Hi @markpope! The benefits are that the SDK client will abstract the mqtt layer concepts for you. For instance, you won't need to know how to serialize each MQTT message and figure out if it was telemetry vs a direct method vs a twin update. Additionally, this SDK has will handle refreshing authentication for you (for symmetric key based authentication) and can handle basic network issues by reconnecting for you. On top of all that, you will also get the newest features as soon as they become available on the service side.

The benefits of writing your own MQTT client would be keeping your dependency list smaller, having more control over which MQTT library you use (this SDK only uses [Paho](https://github.com/eclipse/paho.mqtt.java)), and having finer grain control over how messages get queued.

**Reply by timtay-microsoft** [reply]
We generally recommend that first-timers try out the SDK first as writing the required MQTT client code to get the basics up and running is challenging.

**Reply by timtay-microsoft** [reply]
By comparison, all you need to try out this SDK is something like:

```java
DeviceClient deviceClient = new DeviceClient("your connection string", MQTT);
deviceClient.open();
deviceClient.sendEvent(new Message("my telemetry message payload"));
deviceClient.close();

```

**Comment by markpope** [comment]
Thanks for the feedback. I've used the ESP8266 and Java iot libaries from MSFT. I'm really grateful they provide so much cross language support. Since I'm only coding personal projects, I haven't used the other features you mentioned and didn't think of them.
